Lib/site.py

```diff
@@ -444,7 +444,7 @@ def register_readline():
             def write_history():
                 try:
                     readline.write_history_file(history)
-                except (FileNotFoundError, PermissionError):
+                except (FileNotFoundError, PermissionError, OSError):
```

Comment:
This function should not hide all OS errors. The change from bpo-19891 avoids doing that. `FileNotFoundError` and `PermissionError` are subclasses of `OSError` that get raised for specific `errno` values. To generalize this, it should handle `OSError as e` and check whether `isinstance(e, (FileNotFoundError, PermissionError)) or e.errno == -1`. If so, it can `pass` or print an error message to `sys.stderr`, else it should `raise` so that the exception doesn't get ignored.